python/cuml/test/dask/test_coordinate_descent.py

```diff
@@ -0,0 +1,244 @@
+
+        if nrows < 500000:
+            sk_lasso = Lasso(alpha=np.array([alpha]), fit_intercept=True,
+                         normalize=False, max_iter=1000,
+                         selection=algorithm, tol=1e-10)
+            sk_lasso.fit(X, y)
+            sk_predict = sk_lasso.predict(X)
+            sk_r2 = r2_score(y, sk_predict)
+            assert cu_r2 >= sk_r2 - 0.07
+
+    finally:
+        client.close()
+        cluster.close()
+
+
+@pytest.mark.parametrize('datatype', [np.float32, np.float64])
+@pytest.mark.parametrize('column_info', [unit_param([20, 10]),
+                         quality_param([100, 50]),
+                         stress_param([1000, 500])])
+@pytest.mark.parametrize('nrows', [unit_param(500), quality_param(5000),
+                         stress_param(500000)])
+@pytest.mark.parametrize('n_parts', [unit_param(16), quality_param(32),
+                         stress_param(110)])
+def test_lasso_default(datatype, nrows, column_info, n_parts, client=None):
+
+    ncols, n_info = column_info
+    if client is None:
+        cluster = LocalCUDACluster()
+        client = Client(cluster)
+
+    try:
+        from cuml.dask.linear_model import Lasso as cuLasso
+
+        nrows = np.int(nrows)
+        ncols = np.int(ncols)
+        X, y = make_regression_dataset(datatype, nrows, ncols, n_info)
+
+        X_df, y_df = _prep_training_data(client, X, y, n_parts)
+
+        cu_lasso = cuLasso()
+
+        cu_lasso.fit(X_df, y_df)
+        cu_predict = cu_lasso.predict(X_df)
+        cu_r2 = r2_score(y, cu_predict.compute().to_pandas().values)
+
+        sk_lasso = Lasso()
+        sk_lasso.fit(X, y)
+        sk_predict = sk_lasso.predict(X)
+        sk_r2 = r2_score(y, sk_predict)
+        assert cu_r2 >= sk_r2 - 0.07
+
+    finally:
+        client.close()
+        cluster.close()
+
+
+@pytest.mark.parametrize('datatype', [np.float32, np.float64])
+@pytest.mark.parametrize('X_type', ['ndarray'])
+@pytest.mark.parametrize('alpha', [0.2, 0.7])
+@pytest.mark.parametrize('algorithm', ['cyclic', 'random'])
+@pytest.mark.parametrize('nrows', [unit_param(500), quality_param(5000),
+                         stress_param(500000)])
+@pytest.mark.parametrize('column_info', [unit_param([20, 10]),
+                         quality_param([100, 50]),
+                         stress_param([1000, 500])])
+@pytest.mark.parametrize('n_parts', [unit_param(16), quality_param(32),
+                         stress_param(64)])
+def test_elastic_net(datatype, X_type, alpha, algorithm,
+               nrows, column_info, n_parts, client=None):
+    ncols, n_info = column_info
+    
+    ncols, n_info = column_info
+    if client is None:
+        cluster = LocalCUDACluster()
+        client = Client(cluster)
+
+    try:
+        from cuml.dask.linear_model import ElasticNet as cuElasticNet
+
+        nrows = np.int(nrows)
+        ncols = np.int(ncols)
+        X, y = make_regression_dataset(datatype, nrows, ncols, n_info)
```

Comment:
its a very small thing, i was thinking that we can combine the functions: 
`make_regression_dataset` and  `_prep_training_data` and directly return X_df and y_df